src/frontend/src/optimizer/plan_node/logical_agg.rs

```diff
@@ -384,6 +429,17 @@ impl LogicalAgg {
             .any(|call| matches!(call.agg_kind, AggKind::StringAgg | AggKind::ArrayAgg))
     }
 
+    pub(crate) fn can_agg_two_phase(&self) -> bool {
+        self.agg_calls().iter().all(|call| {
+            matches!(
+                call.agg_kind,
+                AggKind::Min | AggKind::Max | AggKind::Sum | AggKind::Count
+            ) && !call.distinct
+            // QUESTION: why do we need `&& call.order_by_fields.is_empty()` ?
+            //    && call.order_by_fields.is_empty()
```

Comment:
> 🤔 ?

I don't know why we need this condition

Reply:
May because agg calls whose results are affected by the user-specified aggregation order are not able to be 2-phased, however in this code, `min`/`max` result is not related to order so the check is useless now.